# Q1. Write a program to print Hello python ten times.

In [3]:
i=1
while i<=10:
    print(i,'hello')
    i=i+1

1 hello
2 hello
3 hello
4 hello
5 hello
6 hello
7 hello
8 hello
9 hello
10 hello


# Q2. write a program to print number from 1 to 10.

In [4]:
i=1
while i<=10:
    print(i)
    i=i+1

1
2
3
4
5
6
7
8
9
10


# Q3.write a program to print sum of first ten number.

In [12]:
add=0
i=1
while i<=10:
    add=add+i
    i=i+1
print(add)

55


# Q4.Write a program to print n number entered by the user. 

In [11]:
num=int(input('Enter the number:- '))
i=1
while i<=num:
    print(i)
    i=i+1

Enter the number:- 15
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15


# Q5.Write a program to print sum of  n numbers.

In [13]:
num=int(input('Enter the number to add :- '))
add=0
i=1
while i<=num:
    add=add+i
    i=i+1
print(add)

Enter the number to add :- 100
5050


# Q6.Write a program to print table of a number.

In [17]:
n = int(input('Enter the no: '))
for i in range(1,11):
    print(n*i)

Enter the no: 10
10
20
30
40
50
60
70
80
90
100


# Q7. Write a program to calculate the factorial of a number

In [8]:
num=int(input('Enter the number to find factorial:- '))
factorial=1
for i in range(1,num + 1):
    factorial = factorial*i
print("The factorial of",num,"is",factorial)

Enter the number to find factorial:- 4
The factorial of 4 is 24


# Q8.Write a program to print even number between 1 to 100.

In [14]:
start=int(input('Enter the starting number:- '))
End=int(input('Enter the ending number:- '))
for i in range(start,End+1):
    if i%2==0:
        print(i,end=' ')
print('These are Even numbers')

Enter the starting number:- 1
Enter the ending number:- 100
2 4 6 8 10 12 14 16 18 20 22 24 26 28 30 32 34 36 38 40 42 44 46 48 50 52 54 56 58 60 62 64 66 68 70 72 74 76 78 80 82 84 86 88 90 92 94 96 98 100 These are Even numbers


# Q9.Write a program to print odd number between 1 to 100.

In [22]:
start=int(input('Enter the starting number:- '))
End=int(input('Enter the ending number:- '))
for i in range(start,End+1):
    if i%2==1:
        print(i,end=' ')
print('These are odd numbers')

Enter the starting number:- 1
Enter the ending number:- 100
1 3 5 7 9 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 51 53 55 57 59 61 63 65 67 69 71 73 75 77 79 81 83 85 87 89 91 93 95 97 99 These are odd numbers


# Q10.Write a program to check whether a given number is palindrome or not.

In [5]:
num = int(input("Enter a number:"))  
temp = num  
rev = 0  
while num > 0:  
    dig = num % 10
    rev = rev * 10 + dig  
    num = num // 10  
if temp == rev:  
    print("This value is a palindrome number!")  
else:  
    print("This value is not a palindrome number!") 

Enter a number:151
This value is a palindrome number!


# Q11.Write a program to find factors of a number.

In [18]:
num=int(input('Enter the number:- '))
for i in range (1,num+1):
    if num % i == 0:
        print(i,end=',')
print('These are factors of',num)

Enter the number:- 4
1,2,4,These are factors of 4


# Q12.Write a program to print reverse of a number.

In [1]:
num = int(input("Enter a number:"))
rev = 0  
for i in range (0,num):
    while num > 0:
        dig = num % 10
        rev = rev * 10 + dig
        num = num // 10 
print('Reverse of given number is',rev)

Enter a number:123456
Reverse of given number is 654321


# Q13.Write a program to print the sum of reverse of a number.

In [2]:
num = int(input("Enter a number:"))
rev = 0  
sum = 0
for i in range (0,num):
    while num > 0:
        dig = num % 10
        rev = rev * 10 + dig
        num = num // 10 
        sum+=dig
#print('Reverse of given number is',rev)
print('sum of reverse number is:-',sum)

Enter a number:-12365
sum of reverse number is:- 0


# Q14.Write a program to check whether a given number is prime or not.


In [21]:
n=int(input('enter number '))
for i in range(2,n):
    if n % i ==0:
        print(n,'is not a prime')
        break
else:
    print('Given number is prime')

enter number11
Given number is prime


# Q15.Write a program to print prime number between 1 to 100.

In [32]:
start=int(input('Enter the starting number:- '))
end=int(input('Enter the ending number:- '))
for Number in range (start,end+1):
    prime = 0
    for i in range(2, (Number//2 + 1)):
        if(Number % i == 0):
            prime = prime + 1
           # break

    if (prime == 0 and Number != 1):
        print(Number,end=' ')
print('These are prime numbers between',start,'to',end)


Enter the starting number:- 1
Enter the ending number:- 100
2 3 5 7 11 13 17 19 23 29 31 37 41 43 47 53 59 61 67 71 73 79 83 89 97 These are prime numbers between 1 to 100


# Q16.Write a program to print number in reverse order entered by the user.

In [33]:
num = int(input("Enter a number:"))
rev = 0  
for i in range (0,num):
    while num > 0:
        dig = num % 10
        rev = rev * 10 + dig
        num = num // 10 
print('Reverse of given number is',rev)

Enter a number:123456
Reverse of given number is 654321


# Q17.Write a program to print the cube of all numbers from 1 to given number

In [40]:
num=int(input('Enter the number:- '))
for i in range (1,num+1):
    i=i*i*i
    print(i,end=',')
print('This is cube of given number')

Enter the number:- 10
1,8,27,64,125,216,343,512,729,1000,This is cube of given number


In [3]:
n = int(input("Enter a number: "))
reverse = 0
while n > 0:
    digit = n % 10
    reverse = reverse * 10 + digit
    n = n // 10
print("The sum of", n, "and its reverse is", n + reverse)

Enter a number: 123
The sum of 0 and its reverse is 321


In [5]:
num = int(input("Enter a number:"))
rev = 0  
sum = 0
for i in range (0,num):
    while num > 0:
        dig = num % 10
        rev = rev * 10 + dig
        num = num // 10 
        sum+=dig
print('Reverse of given number is',rev)
print('sum of reverse number is:-',sum)

Enter a number:123
Reverse of given number is 321
sum of reverse number is:- 6


# Q18.Write a program to count the total number of digits in a number.

In [1]:
number = int(input("Enter a number: "))
count = 0

for digit in str(number):
    count += 1

print("The number of digits in the entered number is:", count)

Enter a number: 123456789
The number of digits in the entered number is: 9


# Q19.Write a program to print fibonacci series

In [6]:
n = int(input("Enter the number of terms: "))
a, b = 0, 1

if n <= 0:
    print("Invalid input. Please enter a positive integer.")
elif n == 1:
    print("Fibonacci series upto", n, ":")
    print(a)
else:
    print("Fibonacci series:")
    for i in range(n):
        print(a)
        c = a + b
        a, b = b, c

Enter the number of terms: 10
Fibonacci series:
0
1
1
2
3
5
8
13
21
34


# Q20.Writ a program to check whether a given number is amstrong number or not.

In [7]:
num=int(input('Enter the number to check : '))
sum=0
pow=len(str(num))
copy_num=num
while num > 0:
    dig = num % 10
    sum += dig ** pow
    num = num // 10
if sum == copy_num:
    print(f'{copy_num} is an Amstrong number')
    
else:
    print(f'{copy_num} is not an Amstrong number')


Enter the number to check : 542
542 is not an Amstrong number


# Q21.Write a program to check whether a given number is strong number or not.

In [8]:
num = int(input('Enter the number to check: '))
temp=num
sum=0
while num>0:
    dig=num%10
    fact=1
    #print('Digit=',dig)
    for i in range (1,dig+1):
        fact=fact*i
        #print('Fact',fact)
    sum=sum+fact
    #print('Sum',sum)
    num=num//10
if sum == temp:
    print(f'{temp} is a strong number')
else:
    print(f'{temp} is not a strong number')

Enter the number to check: 452
452 is not a strong number


# Q22.Write a program to check whether a given number is perfect number or not


In [9]:
num=int(input('Enter the number'))
sum=0

for i in range(1,num):
    if num % i ==0:
        sum +=i
if sum==num:
    print(f'{num} is a perfect number')
else:
    print(f'{num} is not a perfect number')

Enter the number145
145 is not a perfect number


# Q23.Write a program to find numbers which are divisible by 7 and multiple of 5, between 1000 and 2000.

In [17]:
for num in range(1000,2001):
    if num % 7 ==0 and num % 5 ==0:
        print(num,end=",")
print('   these numbers are divisible by 7 and multiple of 5')

1015,1050,1085,1120,1155,1190,1225,1260,1295,1330,1365,1400,1435,1470,1505,1540,1575,1610,1645,1680,1715,1750,1785,1820,1855,1890,1925,1960,1995,   these numbers are divisible by 7 and multiple of 5


# Q24.Write a program to print the following number pattern using loop
a)1        b)1        c)54321   d)12345  e)5 4 3 2 1
  12         22         5432      1234     4 3 2 1
  123        333        543       123      3 2 1
  1234       4444       54        12       2 1
  12345      55555      5         1        1

In [18]:
#A)

for i in range(1, 6):
    for j in range(1, i+1):
        print(j, end='')
    print()

1
12
123
1234
12345


In [19]:
# B)

for i in range(1, 6):
    for j in range(1, i+1):
        print(i, end='')
    print()

1
22
333
4444
55555


In [20]:
# C)

for i in range(5, 0, -1):
    for j in range(1, i+1):
        print(j, end='')
    print()

12345
1234
123
12
1


In [21]:
# D)

for i in range(5,0,-1):
    for j in range(1, i+1):
        print(j, end='')
    print()


12345
1234
123
12
1


In [22]:
# E)

for i in range(5, 0, -1):
    for j in range(i, 0, -1):
        print(j, end=' ')
    print()


5 4 3 2 1 
4 3 2 1 
3 2 1 
2 1 
1 


# Q25.Write a program to print the following star pattern using a loop
a)*       b)*****   c)*****   d)*****    e)    *
  **        *****     ****       ****         **
  ***       *****     ***         ***        ***
  ****      *****     **           **       ****
  *****     *****     *             *      *****   

In [23]:
# A)

for i in range(6):
    for j in range(i):
        print('*',end='')
    print()


*
**
***
****
*****


In [24]:
# B)

for i in range(6):
    for j in range(6):
        print('*',end='')
    print()

******
******
******
******
******
******


In [25]:
# C)

for i in range(5,0,-1):
    for j in range(1,i+1):
        print('*',end='')
    print()

*****
****
***
**
*


In [26]:
# D)

for i in range(5):
    for j in range(i):
        print(' ', end='')
    for k in range(5-i):
        print('*', end='')
    print()


*****
 ****
  ***
   **
    *


In [27]:
# E)

for i in range(5,0,-1):
    for j in range(i):
        print(' ',end='')
    for k in range(6-i):
        print('*',end='')
    print()

     *
    **
   ***
  ****
 *****


# Q26.Write a program to print the following alphabet pattern using a loop.
a)A       b)A     c)ABCDE    d)E E E E E
  AB        BB      ABCD        D D D D
  ABC       CCC     ABC          C C C
  ABCD      DDDD    AB            B B
  ABCDE     EEEEE   A              A

In [28]:
# A)

n = 5
for i in range(n):
    for j in range(i+1):
        print(chr(65+j), end='')
    print()

A
AB
ABC
ABCD
ABCDE


In [29]:
# B)

n = 5
for i in range(n):
    for j in range(i+1):
        print(chr(65+i), end='')
    print()

A
BB
CCC
DDDD
EEEEE


In [30]:
# C)

for i in range(5):
    for j in range(5-i):
        print(chr(65+j), end='')
    print()


ABCDE
ABCD
ABC
AB
A


In [31]:
# E)

n = 5
for i in range(n):
    print(' ' * (i*2), end='')
    for j in range(n-i):
        print(chr(65+i), end='  ')
    print()


A  A  A  A  A  
  B  B  B  B  
    C  C  C  
      D  D  
        E  
